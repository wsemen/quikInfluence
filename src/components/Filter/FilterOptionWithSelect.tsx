import {
  AccordionPanel,
  Flex,
  Tag,
  TagCloseButton,
} from '@chakra-ui/react';
import DropdownSelect from 'components/DropdownSelect';
import React, { useEffect, useState } from 'react';
import { FilterItemProps } from 'types';
import CustomAccordionButton from './AccordionButton';

const FilterOptionWithSelect = ({ isExpanded, onChange, title = '', selectOptions }: FilterItemProps) => {
  const [selectedOpt, setSelectedAge] = useState<Array<string>>([]);

  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    if (!selectedOpt.includes(e.target.value)) {
      setSelectedAge([...selectedOpt, e.target.value]);
    }
  };

  const handleRemoveOpt = (removeOpt: string) => {
    setSelectedAge(selectedOpt.filter(opt => opt !== removeOpt));
  };

  useEffect(() => {
    onChange({ [title]: selectedOpt });
  }, [selectedOpt]);

  return (
    <>
      <CustomAccordionButton label={title} isExpanded={isExpanded} />
      <AccordionPanel pl="unset" pb={4}>
        <DropdownSelect
          onChange={handleChange}
          options={selectOptions || []}
        />
        <Flex flexWrap="wrap" spacing={4} mt={5}>
          {selectedOpt &&
            selectedOpt.map((opt, index) => (
              <Tag
                size="lg"
                key={`${opt}_${index}`}
                variant="solid"
                borderRadius="full"
                colorScheme="blackAlpha"
              >
                {opt}
                <TagCloseButton data-test-id="remove-opt" onClick={() => handleRemoveOpt(opt)} />
              </Tag>
            ))}
        </Flex>
      </AccordionPanel>
    </>
  );
};

export default FilterOptionWithSelect;
