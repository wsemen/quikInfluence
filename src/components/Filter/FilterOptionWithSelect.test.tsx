import { Accordion, AccordionItem } from "@chakra-ui/react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import FilterOptionWithSelect from "./FilterOptionWithSelect";

it("renders FilterOptionWithSelect component", async () => {
  const mockedOnChange = jest.fn();
  const selectOptions = [
    { label: "Active", value: "active" },
    { label: "Inactive", value: "inactive" },
    { label: "Incomplete", value: "incomplete" },
    { label: "Complete", value: "complete" }
  ];
  const { container } = render(
    <Accordion>
      <AccordionItem border="none">
        {({ isExpanded }) =>
          <FilterOptionWithSelect
            title="test"
            onChange={mockedOnChange}
            isExpanded={true}
            selectOptions={selectOptions}
            data-test-id="select-component"
          />}
      </AccordionItem>
    </Accordion>
  );
  expect(container).toMatchSnapshot();
});
