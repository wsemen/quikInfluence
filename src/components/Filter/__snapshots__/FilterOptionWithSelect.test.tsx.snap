// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FilterOptionWithSelect component 1`] = `
<div>
  <div
    class="chakra-accordion css-0"
  >
    <div
      class="chakra-accordion__item css-cn9ye4"
    >
      <h2>
        <button
          aria-controls="accordion-panel-2"
          aria-expanded="false"
          class="chakra-accordion__button css-12ttzao"
          data-index="0"
          id="accordion-button-2"
          type="button"
        >
          <div
            class="css-8tcdg7"
          >
            test
          </div>
          <svg
            class="chakra-icon css-1hiix9"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g
              fill="currentColor"
            >
              <rect
                height="4"
                width="20"
                x="2"
                y="10"
              />
            </g>
          </svg>
        </button>
      </h2>
      <div
        class="chakra-collapse"
        style="overflow: hidden; display: none; opacity: 0; height: 0px;"
      >
        <div
          aria-labelledby="accordion-button-2"
          class="chakra-accordion__panel css-4rbilf"
          id="accordion-panel-2"
          role="region"
        >
          <div
            class="chakra-form-control css-0"
            role="group"
          >
            <div
              class="chakra-select__wrapper css-42b2qy"
            >
              <select
                class="chakra-select css-1ik61og"
                data-test-id="select-component"
                id="country"
              >
                <option
                  value=""
                >
                  Select ---
                </option>
                <option
                  data-testid="select-option"
                  value="active"
                >
                  Active
                </option>
                <option
                  data-testid="select-option"
                  value="inactive"
                >
                  Inactive
                </option>
                <option
                  data-testid="select-option"
                  value="incomplete"
                >
                  Incomplete
                </option>
                <option
                  data-testid="select-option"
                  value="complete"
                >
                  Complete
                </option>
              </select>
              <div
                class="chakra-select__icon-wrapper css-162mkon"
              >
                <svg
                  aria-hidden="true"
                  class="chakra-select__icon"
                  focusable="false"
                  role="presentation"
                  style="width: 1em; height: 1em; color: currentColor;"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="css-bkneeg"
            spacing="4"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
